magento/magento2
# Fix/24442 images not loaded from media storage

### Description (*)
#24442 The media storage uses the media directory as base directory in syncing (Files in the storage database doesn't contain media/ as directory prefix).
So i changed the base directory for synchronization to the media directory, and passes the files relative to this base path to the constructor in the get.php
#24446 Error occurs during resizing of wysiwyg images with product/catalog config, so validate if resizing is possible
#24445 One off both path's contains a / so remove slashes before during comparison

### Fixed Issues (if relevant)
1. fixes magento/magento2#24442: Since Magento 2.3 the wysiwyg images are not loaded anymore from media storage database.
2. fixes magento/magento2#24445: var/resource_config.json is regenerated each time an image is requested by get.php (media/wysiswig/image.jpg)
3. fixes magento/magento2#24446: First time wysiwyg image is loaded by media storage, placeholder image is shown.

### Manual testing scenarios (*)
1. Fresh install of magento with sample data
2. Synchronize media files to datase, according the manual (https://docs.magento.com/m2/ce/user_guide/system/media-storage-database.html)
3. Add following record to core_config_data table
    - scope: default
    - scope_id: 0
    - path: web/seo/use_rewrites
    - value: 1
4. Remove an image from media/wysiwyg editor directory
5. Load it on the front-end (according issue #24446) the first time you see a placeholder image( instead of 404 page).  After refreshing you see the image.

### Questions or comments
not applicable

### Contribution checklist (*)
